### PR TITLE
Tic Tac Toe 

### PR DESCRIPTION
**Pull request by josepplloo**
## Summary

- Intro to React tutorial completed.

## Screenshots

tictactoe

## Trello Card

https://trello.com/c/Gv6mNt4E/42-react-app-tic-tac-toe

### CODE REVIEW

File: react/tic-tac-toe/src/app/screens/Game/components/Board/index.js [hunk 1]
```diff
@@ -5,8 +5,8 @@ import Square from '../Square';
 import styles from './styles.module.scss';
 
 class Board extends Component {
-  renderSquare() {
-    return <Square />;
+  renderSquare(i) {
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Board/index.js]: I suggest use a most descriptive name for variable i

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
 import styles from './styles.module.scss';
 
-const Square = () => (
+const Square = props => (
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: Here you can receive value with destructuring const Square = ({value}) => (...)

File: react/tic-tac-toe/src/app/screens/Game/components/Board/index.js [hunk 3]
```diff
@@ -5,8 +5,8 @@ import Square from '../Square';
 import styles from './styles.module.scss';
 
 class Board extends Component {
-  renderSquare() {
-    return <Square />;
+  renderSquare(id) {
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Board/index.js]: Here at wolox we try to use arrow function for this type of functions 
`renderSquare = (id) => <Square id={id} />`

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 4]
```diff
@@ -1,11 +1,32 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: Here you export all prop-types but only use number, I recommend use something like:
```import { number } from 'prop-types```;

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 5]
```diff
@@ -1,11 +1,32 @@
...
+    );
+  }
+}
+
+Square.handleClick = () => {
+  this.setState({ value: 'X' });
+};
+
+Square.propTypes = {
+  id: PropTypes.number
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: So if you use the second comment here just only use ```id: number```

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 6]
```diff
@@ -1,11 +1,32 @@
...
+  render() {
+    return (
+      <button key={this.props.id} type="button" className={styles.square} onClick={this.handleClick}>
+        {this.state.value}
+      </button>
+    );
+  }
+}
+
+Square.handleClick = () => {
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: I don't understand this function here???

File: react/tic-tac-toe/src/app/screens/Game/components/Board/index.js [hunk 7]
```diff
@@ -5,12 +5,22 @@ import Square from '../Square';
...
-  }
+  state = {
+    squares: Array(9).fill(null),
+    xIsNext: true
+  };
+
+  handleClick = id => {
+    const { squares, xIsNext } = this.state;
+    const slicedSquares = squares.slice();
+    slicedSquares[id] = this.state.xIsNext ? 'X' : 'O';
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Board/index.js]: Change this.state.xIsNext for xIsNext that you already destructuring it.

File: react/tic-tac-toe/src/app/screens/Game/components/Board/index.js [hunk 8]
```diff
@@ -1,19 +1,16 @@
...
 
 import Square from '../Square';
 
 import styles from './styles.module.scss';
 
 class Board extends Component {
-  renderSquare() {
-    return <Square />;
-  }
+  renderSquare = id => <Square id={id} value={this.props.squares[id]} onClick={this.props.onClick} />;
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Board/index.js]: Here you can use destructuring of props.

File: react/tic-tac-toe/src/app/screens/Game/components/Square/index.js [hunk 9]
```diff
@@ -1,11 +1,28 @@
...
+    return (
+      <button type="button" className={styles.square} onClick={this.handleClick}>
+        {value}
+      </button>
+    );
+  }
+}
+
+Square.propTypes = {
+  id: number,
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/components/Square/index.js]: Same of previous comment

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 10]
```diff
@@ -1,20 +1,87 @@
...
+
+    squares[id] = xIsNext ? 'X' : 'O';
+    this.setState({
+      history: historyPoint.concat([{ squares }]),
+      xIsNext: !xIsNext,
+      stepNumber: historyPoint.length
+    });
+  };
+
+  handleMove = move => {
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: This can be on one line

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 11]
```diff
@@ -1,20 +1,87 @@
...
+        <li key={moveKey}>
+          <Moves desc={desc} onClick={this.handleMove} move={move} />
+        </li>
+      );
+    });
+
+    if (winner) {
+      status = `Winner: ${winner}`;
+    } else {
+      status = `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: Use xIsNext on destructuring of this.state.

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 12]
```diff
@@ -1,20 +1,87 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.module.scss';
```
Comment (cvega08) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: Move styles to end

File: react/tic-tac-toe/src/app/screens/Game/components/Board/index.js (hunk 8 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/app/screens/Game/components/Board/index.js]: No matter if this avoid using more lines?

**Pull request merged**